tools/plugins/com.liferay.ide.ui/src/com/liferay/ide/ui/templates/SuperClassNameResolver.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.ui.templates;
+
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.internal.corext.template.java.CompilationUnitContext;
+import org.eclipse.jface.text.templates.TemplateContext;
+import org.eclipse.jface.text.templates.TemplateVariableResolver;
+
+/**
+ * @author Joye Luo
+ */
+@SuppressWarnings( "restriction" )
+public class SuperClassNameResolver extends TemplateVariableResolver
+{
+
+    public SuperClassNameResolver()
+    {
+        super( "super_class_name", "get superclass name" );
+    }
+
+    @Override
+    protected String resolve( TemplateContext context )
+    {
+        String superClassName = "";
+
+        ICompilationUnit unit = ( (CompilationUnitContext) context ).getCompilationUnit();
```

Comment:
It was excepted to use in liferay module project for this template:
<template name="Component" description="used in liferay module project"
    id="org.eclipse.jdt.ui.templates.Component" context="java"
    enabled="true" autoinsert="false">
   @Component(
	immediate = true,
	property = {
		"com.liferay.portlet.display-category=category.sample",
		"com.liferay.portlet.instanceable=true",
		"javax.portlet.display-name=${primary_type_name} Portlet",
		"javax.portlet.init-param.template-path=/",
		"javax.portlet.init-param.view-template=/view.jsp",
		"javax.portlet.resource-bundle=content.Language",
		"javax.portlet.security-role-ref=power-user,user"
	},
	service = ${superclass_name}.class
	)
  </template>